substrate/frame/staking-async/src/tests/slashing.rs

```diff
@@ -984,37 +984,56 @@ fn remove_deferred() {
 
 		// fails if empty
 		assert_noop!(
-			Staking::cancel_deferred_slash(RuntimeOrigin::root(), 1, vec![]),
+			Staking::cancel_deferred_slash(RuntimeOrigin::root(), 3, vec![]),
 			Error::<T>::EmptyTargets
 		);
 
-		// cancel the slash with 10%.
+		// cancel the slash with 12% (which covers the 10% slash but not the 15% slash)
 		assert_ok!(Staking::cancel_deferred_slash(
 			RuntimeOrigin::root(),
 			3,
-			vec![(11, Perbill::from_percent(10), 0)]
+			vec![(11, Perbill::from_percent(12))],
 		));
-		assert_eq!(UnappliedSlashes::<T>::iter_prefix(&3).count(), 1);
+
+		// validator 11 slash is set to be cancelled for era 3 with upto 12% cancellation.
+		assert_eq!(CancelledSlashes::<T>::get(&3), vec![(11, Perbill::from_percent(12))]);
+
 		assert_eq!(
 			staking_events_since_last_call(),
-			vec![Event::SlashCancelled {
-				slash_era: 3,
-				slash_key: (11, Perbill::from_percent(10), 0),
-				payout: 12
-			}]
+			vec![Event::SlashCancelled { slash_era: 3, validator: 11 }]
 		);
 
-		// apply the one with 15%.
+		// roll to the slashing era.
 		Session::roll_until_active_era(3);
+		// slash still exists in unapplied.
+		assert_eq!(UnappliedSlashes::<T>::iter_prefix(&3).count(), 2);
+
+		// clear events from session rotation
 		let _ = staking_events_since_last_call();
+
 		Session::roll_next();
-		assert_eq!(
-			staking_events_since_last_call(),
-			vec![
-				Event::Slashed { staker: 11, amount: 50 },
-				Event::Slashed { staker: 101, amount: 12 }
-			]
-		);
+		// first slash (10%) is cancelled, no events
+		assert_eq!(staking_events_since_last_call(), vec![]);
+		// one slash processed, one more to go
+		assert_eq!(UnappliedSlashes::<T>::iter_prefix(&3).count(), 1);
+
+		Session::roll_next();
+		// second slash (15%) is NOT cancelled, so it gets applied
+		let events = staking_events_since_last_call();
+		assert_eq!(events.len(), 2);
+		// The 15% slash already has computed amounts in the UnappliedSlash
+		// Based on the test setup, it has own: 50, others: [(101, 12)]
+		assert_eq!(events[0], Event::Slashed { staker: 11, amount: 50 });
+		assert_eq!(events[1], Event::Slashed { staker: 101, amount: 12 });
+
+		// check balances after 15% slash
+		assert_eq!(asset::stakeable_balance::<T>(&11), 950);
+		assert_eq!(asset::stakeable_balance::<T>(&101), 488);
+
+		// all unapplied slashes should have been removed
+		assert_eq!(UnappliedSlashes::<T>::iter_prefix(&3).count(), 0);
+		// cancelled slashes should have been cleared after all era slashes are applied.
+		assert_eq!(CancelledSlashes::<T>::get(&3), vec![]);
```

Comment:
Suggest a new test here that demonstrates: 

If there is a bug, and a validator has multiple slashes with different fractions, a single cancellation call from governance with `fraction: 100%` will cancel all of them.

Will it? I think it should, and it is a scenario that I foresee will be useful.